src/lib/OS/SyscallHandler.cc

```diff
@@ -1,11 +1,10 @@
 #include "simeng/OS/SyscallHandler.hh"
 
+#include "simeng/OS/SimOS.hh"
 namespace simeng {
 namespace OS {
 
-SyscallHandler::SyscallHandler(
-    const std::unordered_map<uint64_t, std::shared_ptr<Process>>& processes)
-    : processes_(processes) {
+SyscallHandler::SyscallHandler(SimOS* os) : os_(os) {
```

Comment:
Just a thought on this circular dependancy with SimOS; given that we only need access to `hfmmap_` (currently a unique_ptr) and the `getProcess()` function from SimOS, why don't we construct the syscallHandler by:

- Making `hfmmap_` a shared_ptr and have the syscall handler keep it as a member variable
- Pass getProcess as a callback function

Reply:
Please [see](https://github.com/UoB-HPC/SimEng/pull/283/files#r1111137720)

Reply:
Given that this will be an issue anyway in the short term future, why not change it now? Or would my proposed alternative also not work with SST (assuming we re-implement the getProcess() function in our SimOS replacement component; which is highly likely)

Reply:
If the change is already present in another PR, why should it be handled here? In addition I'm not sure how the change is implemented in the other PR, wont that lead to even more merge conflicts?

Reply:
Sorry, I missed the top comment saying this was being changed in another PR